test/Listener/ConfigListenerTest.php

```diff
@@ -370,4 +370,99 @@ public function testConfigListenerFunctionsAsAggregateListener()
         $configListener->detach($events);
         $this->assertEquals(2, count($this->getEventsFromEventManager($events)));
     }
+
+    public function datasetCachedConfigs()
+    {
+        $datasets = [
+            'valid_file' => [
+                // Test for check if cache is properly used if is valid
+                ['data' => 'any'], // expects to fall back to loading all modules
+                '<?php return [\'data\' => \'any\'];',
+            ],
+            'inexistent_file' => [
+                ['some' => 'thing', 'listener' => 'test'], // expects to fall back to loading all modules
+                null, // file won't be created
+            ],
+            'file_with_data_before_php_tag' => [
+                ['some' => 'thing', 'listener' => 'test'], // expects to fall back to loading all modules
+                'something<?php return [\'data\' => \'any\'];',
+            ],
+            'malformed_open_tag' => [
+                ['some' => 'thing', 'listener' => 'test'], // expects to fall back to loading all modules
+                '<\?php return [\'data\' => \'any\'];',
+            ],
+        ];
+        if (version_compare(PHP_VERSION, '7.0', '>=')) {
+            $datasets['invalid_file_syntax'] = [
+                ['some' => 'thing', 'listener' => 'test'], // expects to fall back to loading all modules
+                '<?php return (???\'data\' => \'any\');',
+            ];
+        }
+        return $datasets;
+    }
+
+    /**
+     * @dataProvider datasetCachedConfigs
+     */
+    public function testDoesNotReturnConfigToOutputIfFileIsMalformed(
+        $expectedConfig,
+        $cacheContents
+    ) {
+        $tempDir = sys_get_temp_dir();
+        $cacheConfigFile = $tempDir . '/module-config-cache.php';
+        $staticConfigFile = $tempDir . '/module-config-static.php';
```

Comment:
If anything is wrong with cached config, 'fallback' mode is run which uses static config to run module manager the same way as there would be no cached file at all.

Reply:
yeah, fine, but in your data provider I can't see that you expected anywhere this file to be used

Reply:
Hm, you are right. I will check it. :)

Reply:
You were right, this test was not depending on static config file. I have removed mocking this file.

Reply:
So, my question is now if it is right? Should we silently skip invalid files and still load the configuration for modules or rather throw an exception when we can't load config file properly?

Reply:
From my perspective - it is right as application won't crash (will silently fall back into gathering all configurations from modules).
Cache will be recreated if it's malformed as result of file being broken.
We have created fork 2 weeks ago we have production-proven that it helps, we have 0 errors like
```
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to Zend\ModuleManager\Listener\ConfigListener::setMergedConfig() must be of the type array, integer given
```
since then.

This changes affect cache file *only* so support for config files stored inside modules is not changed (they will throw errors and it's OK as it will mean that application is somehow broken).

Reply:
@webimpress do you have any doubts? Can i resolve this conversation? :)

Reply:
@rutek I'll test it and let you know. Sorry, I am quite busy recently... Will try to do it in the next days.

Reply:
@rutek Are you on [slack](https://zendframework.slack.com/)? ([Signup for Slack here](https://zendframework-slack.herokuapp.com/)
Please ping me there, I'd like to talk more about it when you have a moment.